server/utils/azureAppInsights.ts

```diff
@@ -29,7 +31,29 @@ export function buildAppInsightsClient(name = defaultName()): TelemetryClient {
   if (process.env.APPINSIGHTS_INSTRUMENTATIONKEY) {
     defaultClient.context.tags['ai.cloud.role'] = name
     defaultClient.context.tags['ai.application.ver'] = version()
+    defaultClient.addTelemetryProcessor(addUserDataToRequests)
     return defaultClient
   }
   return null
 }
+
+export function addUserDataToRequests(
+  envelope: EnvelopeTelemetry,
+  contextObjects: RequestOptions | ClientRequest | Error
+) {
+  const isRequest = envelope.data.baseType === Contracts.TelemetryTypeString.Request
+  if (isRequest) {
+    const { username, activeCaseLoadId, activeCaseLoad, isReceptionUser } =
+      contextObjects?.['http.ServerRequest']?.res?.locals?.user || {}
+    const { properties } = envelope.data.baseData
+    // eslint-disable-next-line no-param-reassign
+    envelope.data.baseData.properties = {
+      username,
+      activeCaseLoadId,
+      activeCaseLoadDescription: activeCaseLoad?.description,
+      isReceptionUser,
+      ...properties,
```

Comment:
Not sure if this is a bit too much info, thinking in terms of PII. I had a quick google and can't seem to say whether it is or isn't, seems fairly subjective in terms of what data could identify someone. They seem like useful data points though whilst we're rolling out

Reply:
think that's fine but can we lose the prison description - quite a lot of additional data storage if you thinks it for every single request and its trivial to do the look up externally.

Reply:
good to add the is reception user bit though - it will allow us to see if its regularly useful info for non-reception staff which is just an assumption at the moment